woocommerce/woocommerce
# Take screenshots of failed e2e tests

### All Submissions:

* [x ] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x ] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x ] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?





### Changes proposed in this Pull Request:

This adds screenshots for broken e2e tests. The screenshots will also be uploaded to an Amazon S3 instance from Travis CI so they can be viewed. If run locally, there will be a screenshots directory created where they can be viewed

If there is a failed e2e test you'll see a block like this in the build in Travis CI that will give the URLS for the images.
<img width="1242" alt="2018-05-30_1213" src="https://user-images.githubusercontent.com/31110506/40739378-3b671b0a-6403-11e8-9de2-7d0b72fca5d4.png">


### How to test the changes in this Pull Request:

1. Run a build in Travis CI with a failing e2e test
2. Ensure the build takes a screenshot and gives and S3 link to view the screenshot

### Other information:

* [x ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x ] Have you written new tests for your changes, as applicable?
* [ x] Have you successfully ran tests with your changes locally?
